src/adapters/aol.js

```diff
@@ -1,183 +1,141 @@
-   */
-  function _addErrorBid(response, context) {
-    var bid = bidsMap[context.alias];
+  function _addBidResponse(bid, response) {
+    var bidData;
 
-    if (!bid) {
-      utils.logError('mismatched bid: ' + context.placement, ADTECH_BIDDER_NAME, context);
+    try {
+      bidData = response.seatbid[0].bid[0];
+    } catch (e) {
+      utils.logError('Invalid bid response', BIDDER_CODE, bid);
+      _addErrorBidResponse(bid, response);
       return;
     }
 
-    // clean up--we no longer need to store the bid
-    delete bidsMap[context.alias];
+    var cpm;
 
-    var bidResponse = bidfactory.createBid(2);
-    bidResponse.bidderCode = ADTECH_BIDDER_NAME;
-    bidResponse.reason = response.getNbr();
-    bidResponse.raw = response.getResponse();
-    bidmanager.addBidResponse(bid.placementCode, bidResponse);
-  }
+    if (bidData.ext && bidData.ext.encp) {
+      cpm = bidData.ext.encp;
+    } else {
+      cpm = bidData.price;
 
-  /**
-   * @private map a prebid bidrequest to an ADTECH/aol bid request
-   * @param {Bid} bid the bid request
-   * @return {Object} the bid request, formatted for the ADTECH/DAC api
-   */
-  function _mapUnit(bid) {
-    var alias = bid.params.alias || utils.getUniqueIdentifierStr();
-
-    // save the bid
-    bidsMap[alias] = bid;
-
-    return {
-      adContainerId: _dummyUnit(bid.params.adContainerId),
-      server: bid.params.server, // By default, DAC.js will use the US region endpoint (adserver.adtechus.com)
-      sizeid: bid.params.sizeId || 0,
-      pageid: bid.params.pageId,
-      secure: document.location.protocol === 'https:',
-      serviceType: 'pubapi',
-      performScreenDetection: false,
-      alias: alias,
-      network: bid.params.network,
-      placement: parseInt(bid.params.placement),
-      gpt: {
-        adUnitPath: bid.params.adUnitPath || bid.placementCode,
-        size: bid.params.size || (bid.sizes || [])[0]
-      },
-      params: {
-        cors: 'yes',
-        cmd: 'bid',
-        bidfloor: (typeof bid.params.bidFloor !== "undefined") ? bid.params.bidFloor.toString() : ''
-      },
-      pubApiConfig: ADTECH_PUBAPI_CONFIG,
-      placementCode: bid.placementCode
-    };
-  }
+      if (cpm === null || isNaN(cpm)) {
+        utils.logError('Invalid price in bid response', BIDDER_CODE, bid);
+        _addErrorBidResponse(bid, response);
+        return;
+      }
+    }
 
-  /**
-   * @private once ADTECH is loaded, request bids by
-   * calling ADTECH.loadAd
-   */
-  function _reqBids() {
-    if (!window.ADTECH) {
-      utils.logError('window.ADTECH is not present!', ADTECH_BIDDER_NAME);
-      return;
+    var ad = bidData.adm;
+    if (bidData.ext && bidData.ext.pixels) {
+      ad += bidData.ext.pixels;
     }
 
-    // get the bids
-    utils._each(bids, function (bid) {
-      var bidreq = _mapUnit(bid);
-      window.ADTECH.loadAd(bidreq);
-    });
+    var bidResponse = bidfactory.createBid(1);
```

Comment:
Consider setting the bid response ID to match the bid request ID as described here: https://github.com/prebid/Prebid.js/issues/509. 

Doing so will position the adapter to take advantage of Prebid Roadmap features such as secure iframe support, improved analytics, Prebid Auctions and so on.